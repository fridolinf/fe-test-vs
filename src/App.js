import "./App.css";
import Avatar from "./components/avatar";

function App() {
  return (
    <div title="app-wrapper" className="App">
      <Avatar size="xs" maxLength={8} />
    </div>
  );
}

export default App;
