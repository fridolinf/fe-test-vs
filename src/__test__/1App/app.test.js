import { screen, render } from "@testing-library/react";
import App from "../../App";
import renderer from "react-test-renderer";
const AppWrapper = () => {
  render(<App />);
};
test("Default component has rendered in app", async () => {
  AppWrapper(); //render component

  const firstDiv = screen.getByTitle(/app-wrapper/i); //get first div component by title="app-wrapper"
  const avatar = screen.getByTitle(/avatar-first-wrapper/i); //get avatar component by title="avatar-first-wrapper"

  expect(firstDiv).toBeInTheDocument(); //expect first div component to render in document
  expect(avatar).toBeInTheDocument(); //expect avatar component to render in document
});

test("Matches DOM Snapshot", () => {
  const domTree = renderer.create(<App />); //create snapshot
  expect(domTree).toMatchSnapshot(); //expect create snapshot file
});
