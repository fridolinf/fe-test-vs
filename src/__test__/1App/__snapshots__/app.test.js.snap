// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches DOM Snapshot 1`] = `
<div
  className="App"
  title="app-wrapper"
>
  <div
    className="avatar-group"
    title="avatar-first-wrapper"
  >
    <div
      className="img-wrapper"
    >
      <img
        alt="To"
        aria-label="avatar-img"
        className="avatar-img xs"
        src="https://cdn-icons-png.flaticon.com/512/1151/1151733.png?w=740&t=st=1676606321~exp=1676606921~hmac=6f18b4f33defc5411880cbda1667fd766caa6a731aa9cb9e3dbd3e2ab67b6872"
      />
    </div>
    <div
      className="img-wrapper"
    >
      <img
        alt="To"
        aria-label="avatar-img"
        className="avatar-img xs"
        src="https://cdn-icons-png.flaticon.com/512/1151/1151733.png?w=740&t=st=1676606321~exp=1676606921~hmac=6f18b4f33defc5411880cbda1667fd766caa6a731aa9cb9e3dbd3e2ab67b6872"
      />
    </div>
    <div
      className="img-wrapper"
    >
      <img
        alt="To"
        aria-label="avatar-img"
        className="avatar-img xs"
        src="https://cdn-icons-png.flaticon.com/512/1151/1151733.png?w=740&t=st=1676606321~exp=1676606921~hmac=6f18b4f33defc5411880cbda1667fd766caa6a731aa9cb9e3dbd3e2ab67b6872"
      />
    </div>
    <div
      className="img-wrapper"
    >
      <div
        className="avatar-img-max xs"
        title="count-wrapper"
      >
        <p
          className="count-max"
          title="count-max"
        >
          +
          5
        </p>
      </div>
    </div>
  </div>
</div>
`;
